app/models/user.rb

```diff
@@ -4,9 +4,17 @@ class User < ApplicationRecord
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable
   has_one :cart
+  has_many :orders
 
-  def admin?
-    print "ROLE",role=="ADMIN"
-    role == "ADMIN"
+  ROLES = {
+    "admin" => "ADMIN",
+    "user" => "USER"
+  }
+
+  def has_role?(role)
+    mapped_role = ROLES[role]
```

Comment:
Thsi method can be simplified as

def has_role?(role)
  return true if self.role == ROLES[role]
  false
end

Reply:
But will this validate if invalid role type is passed ?